generate-rules.sh

```diff
@@ -1,25 +1,58 @@
 #!/usr/bin/env bash
+# dependencies: git, jq, tr (coreutils), awk (gawk)
 
+for c in git jq tr awk; do
+  command -v $c > /dev/null
+  if [[ $? > 0 ]]; then
+    echo "Error: Command $c not found in \$PATH."
+    exit 1
+  fi
+done
+
+tohex() {
+  printf $1 | awk '{ printf("%04x", $1) }'
+}
+
+shopt -s globstar
 set -eu
 
-SRC_ROOT=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
-PROJECT="${SRC_ROOT}/OpenTabletDriver.Tools.udev"
-
-if [ $# -eq 2 ]; then
-    TABLET_CONFIGURATIONS="${1}"
-    RULES_FILE="${2}"
-elif [ $# -eq 0 ]; then
-    TABLET_CONFIGURATIONS="${SRC_ROOT}/OpenTabletDriver.Configurations/Configurations"
-    RULES_FILE="-"
-elif [ $# -ne 2 ]; then
-    echo "Usage: ${0} <configuration folder> <output file>"
-    exit 1
-fi
+OTD_CONFIGURATIONS="${OTD_CONFIGURATIONS:="$(git rev-parse --show-toplevel)/OpenTabletDriver.Configurations/Configurations"}"
+
+script='[
+  .[] | { Name:.Name, libinput:(.Attributes.libinputoverride // "0") } + (.DigitizerIdentifiers[] | { VendorID:.VendorID, ProductID:.ProductID })
+] | unique | sort_by(.VendorID,.ProductID) | .[] | "\(.Name):\(.VendorID):\(.ProductID):\(.libinput)"'
```

Comment:
```suggestion
script='[
  .[] | { Name:.Name, libinput:(.Attributes.libinputoverride // "0") } + (.DigitizerIdentifiers[] | { VendorID:.VendorID, ProductID:.ProductID })
] | unique | sort_by(.VendorID,.ProductID) | group_by(.VendorID, .ProductID) |
map({ Name: (map(.Name) | join(", ")), libinput: .[0].libinput, VendorID: .[0].VendorID, ProductID: .[0].ProductID})
| .[] | "\(.Name):\(.VendorID):\(.ProductID):\(.libinput)"'
```

this groups names in comments properly